File: phase1/src/main/InvertedIndex.java
package main;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class InvertedIndex {
    private final String directoryPath = new File("").getAbsolutePath() + "\\phase1\\database";
    private final HashMap<String, List<String>> invertedIndex = new HashMap<>();


    private void addWordToInvertedIndex(String word, String fileName) {

        List<String> files = invertedIndex.get(word);
        if (files == null) {
            List<String> fileNames = new ArrayList<>();
            fileNames.add(fileName);
            invertedIndex.put(word, fileNames);
            return;
        } else {
            for (String file : files) {
                if (file.equals(fileName)) {
                    return;
                }

            }
        }
        files.add(fileName);


    }

    public HashMap<String, List<String>> makeInvertedIndex() throws IOException {
        HashMap<String, List<String>> wordsInFiles = new FileReader().readDocs(directoryPath);
        for (String fileName : wordsInFiles.keySet()) {
            List<String> words = wordsInFiles.get(fileName);
            for (String word : words) {
                addWordToInvertedIndex(word, fileName);
            }
        }

        return invertedIndex;
    }
}

File: phase5/SimpleSearch/InvertedIndex.cs
namespace SimpleSearch
{
    public class InvertedIndex
    {
        public Dictionary<string, List<string>> Index { get; set; }
        public InvertedIndex()
        {
            Index = new Dictionary<string, List<string>>();
        }
        private void AddWordToInvertedIndex(string word, string fileName)
        {

            var files = Index.FirstOrDefault(a =>a.Key == word).Value;
            if (files == null)
            {
                var fileNames = new List<string>
                {
                    fileName
                };
                Index.Add(word, fileNames);
                return;
            }
            else
            {
                foreach (var file in files)
                {
                    if (file == fileName)
                    {
                        return;
                    }

                }
            }
            files.Add(fileName);

        }
        public Dictionary<string, List<string>> MakeInvertedIndex(Dictionary<string, List<string>> words_in_files)
        {
            foreach (var fileName in words_in_files.Keys)
            {
                var words = words_in_files.First(a => a.Key == fileName).Value;
                foreach (var word in words)
                {
                    AddWordToInvertedIndex(word, fileName);
                }
            }
            return Index;
        }
    }
}